moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp

```diff
@@ -49,17 +49,17 @@ const std::string planning_pipeline::PlanningPipeline::MOTION_PLAN_REQUEST_TOPIC
 const std::string planning_pipeline::PlanningPipeline::MOTION_CONTACTS_TOPIC = "display_contacts";
 
 planning_pipeline::PlanningPipeline::PlanningPipeline(const moveit::core::RobotModelConstPtr& model,
-                                                      const ros::NodeHandle& nh,
+                                                      const ros::NodeHandle& pipeline_nh,
```

Comment:
So with `pipeline_nh_` defaulting to `ros::NodeHandle("~")` and `private_nh_` defaulting to `ros::NodeHandle("~")`, what's the difference? What am I missing?-)

Should the default be different? _Could_ the default be different?

Reply:
MoveGroup or better MoveItCpp constructs multiple planning pipelines with child node handles in order to separate pipeline and plugin parameters. That one shouldn't be used for the topics, though, and that's why there is now a separate node handle which always uses the private "move_group" namespace. The default is not used at all afaik, but technically you could initialize a pipeline with a both node handles pointing to the private namespace.